test/k23si/3SITxnTest.cpp

```diff
@@ -1279,6 +1285,484 @@ seastar::future<> testScenario01() {
+                    .then([](auto&& response) {
+                        auto& [status, resp] = response;
+                        K2EXPECT(status, Statuses::S200_OK);               
+                        K2INFO("SC02.case08-09-10(end an unread key)::OP_END. " << "status: " << status.code << " with MESG: " << status.message);
+                    });
+                })
+                .then([this, &k6, &eqMTR] {
+                    dto::K23SI_MTR bfMTR{
+                        .txnid = txnids++, 
+                        .timestamp = {(eqMTR.timestamp.tEndTSECount() - 500000000), 123, 1000}, // 500ms earlier
+                        .priority = dto::TxnPriority::Medium};
+                    return doRead<DataRec>(k6, bfMTR, collname, ErrorCaseOpt::NoInjection)
+                    .then([](auto&& response) {
+                        auto& [status, resp] = response;
+                        K2INFO("SC02.case08-09-10(before)::OP_READ. " << "status: " << status.code << " with MESG: " << status.message);
+                        K2EXPECT(status, Statuses::S404_Not_Found);               
+                    });
+                }) // end sc-02 case-08
+                .then([this, &k6, &eqMTR, &v1] {
+                    return doRead<DataRec>(k6, eqMTR, collname, ErrorCaseOpt::NoInjection)
+                    .then([&v1](auto&& response) {
+                        auto& [status, resp] = response;
+                        K2INFO("SC02.case08-09-10(equal)::OP_READ. " << "status: " << status.code << " with MESG: " << status.message);                        
+                        K2INFO("Value of Key6: " << resp.value.val);
+                        K2EXPECT(status, Statuses::S200_OK);
+                        K2EXPECT(resp.value.val, v1);
+                    });
+                }) // end sc-02 case-09
+                .then([this, &k6, &eqMTR, &v1] {
+                    dto::K23SI_MTR afMTR{
+                        .txnid = txnids++, 
+                        .timestamp = {(eqMTR.timestamp.tEndTSECount() + 500000000), 123, 1000}, // 500ms later
+                        .priority = dto::TxnPriority::Medium};
+                    return doRead<DataRec>(k6, afMTR, collname, ErrorCaseOpt::NoInjection)
+                    .then([&v1](auto&& response) {
+                        auto& [status, resp] = response;
+                        K2INFO("SC02.case08-09-10(after)::OP_READ. " << "status: " << status.code << " with MESG: " << status.message);
+                        K2INFO("Value of Key6: " << resp.value.val);
+                        K2EXPECT(status, Statuses::S200_OK);                        
+                        K2EXPECT(resp.value.val, v1);
+                    });
+                }); // end sc-02 case-10
+            }
+        );
+    }) // end sc-02 case-08-09-10
+    .then([] {
+    // SC02 case11 & 12 & 13 & 14: Async END test (end-but-not-finalized key )
```

Comment:
Can you explain a bit more in the comment what the purpose of this test case is?